Automattic/jetpack
# Protect: render Security component with dynamic data

This PR removes the hardcoded Security bundle data, replacing them with data provided by wpcom side.
It makes the implementation more reliable and consistent. Easier to maintain.

In a previous PR, we've exposed the `securityBundle` by the `useProtectData()` hook. The `securityBundle` is propagated from the backend to the client via the `jetpackProtectInitialState` global var. This data is requested to the wpcom side, hitting a proper endpoint.

Fixes #

#### Changes proposed in this Pull Request:

* Protect: render Security component with dynamic data.
* Clean mock data
* Improve the Interstitial story adding mock data.

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?



#### Testing instructions:






Test with Storybook
* Go to Interstitial story http://localhost:50240/?path=/story/plugins-protect-interstitial--default
* Confirm data there is accurate. Keep in mind it uses mocked data susceptible to possible changes.

<img width="984" alt="image" src="https://user-images.githubusercontent.com/77539/167161577-773a87af-1b0a-4c9d-9d60-8dae75c4f0c3.png">

**Test in Protect overview**
* Disconnect your testing site to see the interstitial page
* Confirm you see the Security card properly

<img width="1161" alt="image" src="https://user-images.githubusercontent.com/77539/167161810-27709538-c47b-44d4-a12f-8651dbc79194.png">

before | after | pricing
------|------|------
<img width="465" alt="Screen Shot 2022-05-06 at 12 14 20 PM" src="https://user-images.githubusercontent.com/77539/167162000-00682ba8-1075-4e17-8622-6952259774b4.png"> | <img width="461" alt="Screen Shot 2022-05-06 at 12 15 29 PM" src="https://user-images.githubusercontent.com/77539/167162134-a95a4f53-87f6-4c1b-94b4-900c302485f5.png"> | <img width="418" alt="image" src="https://user-images.githubusercontent.com/77539/167163449-4b08a5a4-f22d-418f-9b06-8a01f375fe22.png">



---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202232739997899
